### PR TITLE
changes the default message digest / algorithm from sha1 to sha256

### PR DESCRIPTION
**Pull request by pcabido**
SHA1 should no longer be a default anywhere. This changes PR changes the default message digest / algorithm from sha1 to sha256.
SHA1 is still available in the KNOWN_MDS, if anyone wants to use it they still can, just not by mistake.

### CODE REVIEW

**Comment by reaperhulk** [comment]
Looks like the tests will need to be updated to expect sha256. I'll be happy to merge this once the tests are updated!

**Comment by pcabido** [comment]
@reaperhulk I can definitely do that, should have checked in the first place. A lot more need to be updated before you should merged this PR. Let me take care of it.

**Comment by pcabido** [comment]
@reaperhulk can you take a look? I'm using the quay.io/travisci/travis-ruby image to run the tests locally and I can't reproduce this failure.

**Comment by reaperhulk** [comment]
@pcabido I think we may have a bit rot problem with CI here. I'm going to try submitting an empty PR to see if it fails.

**Comment by reaperhulk** [comment]
@pcabido would you mind merging master or rebasing this with the updated travis config?

File: spec/csr_spec.rb [hunk 1]
```diff
@@ -285,7 +290,7 @@
   it "loads a csr with load_from_file" do
     path = File.dirname(__FILE__) + '/fixtures/csr1.pem'
     csr = R509::CSR.load_from_file path
-    expect(csr.message_digest.name).to eq('sha1')
+    expect(csr.message_digest.name).to eq('sha256')
```
Comment (reaperhulk) [inline spec/csr_spec.rb]: What's the reason for changing csr1? It's just a test fixture. Were there no sha256 CSRs?

Comment (pcabido) [inline spec/csr_spec.rb]: Yes, I thought I'd change the _default_ to sha256 as well as there were no other. I couldn't consistently do this, `cert1.pem` for example I couldn't change without replacing the cert.

Comment (reaperhulk) [inline spec/csr_spec.rb]: Okay, got it. Alright, I think this looks good. Thanks for sticking with this :)

**Pull request merged**